test/fgpu/test_engine.py

```diff
@@ -913,9 +915,60 @@ async def test_missing_heaps(
                 # the time difference is a multiple of the mixer wavelength.
                 np.testing.assert_equal(x, y)
 
+        # Information needed to check that the dig-rms-dbfs sensor is correctly
+        # going into FAILURE when there are missing samples.
+        time_converter = TimeConverter(SYNC_TIME, ADC_SAMPLE_RATE)
+        unix_timestamps = [time_converter.adc_to_unix(t) for t in timestamps]
+        # The sensor is updated once per out_item.
+        timestamps_per_output_chunk = engine_server.chunk_jones // output.channels
+        decimation = output.decimation  # Check if we are in narrowband.
+
         for pol in range(N_POLS):
+            # Check prometheus counter
             input_missing_heaps = np.sum(~src_present[pol])
             assert prom_diff.get_sample_diff("input_missing_heaps_total", {"pol": str(pol)}) == input_missing_heaps
+
+            # Check dig-rms-dbfs sensor. Not present in the narrowband mode.
+            if decimation == 1:
+                sensor_update_iter = iter(sensor_update_dict[f"input{pol}.dig-rms-dbfs"])
+                sensor_update = next(sensor_update_iter)
+                # The sensor is updated once per out_item. But the step between
+                # timestamps of the sensor update is not consistent, it wobbles a
+                # bit (I am currently not sure why). So we need to keep track of
```

Comment:
Can you elaborate? Is this just the limitations of floating-point rounding (which limits UNIX timestamps to about 200ns precision these days) or something bigger?

Reply:
I am not quite sure whether it's just floating-point rounding or something else.

In the first test which I used for debugging, with a `spectra_per_chunk = 2048` I saw mostly diffs of 2047, 2048, a handful of 2050. And then a really large one (34848) where there was the large gap.

I couldn't see any kind of pattern.

Reply:
In retrospect I think it was floating-point rounding. The individual spectra are close enough to each other that often multiple of them get the same output when converting ADC to unix. So which one was being matched was effectively random.